config/validation.go

```diff
@@ -0,0 +1,86 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+)
+
+// Validate validate the config as an input. If not valid, it returns error
+func Validate(conf Config) error {
```

Comment:
I don't see a need for Config interface and we can validate clientConfig and ServerConfig explicitly where needed, currently can you add an explicit TODO such that we will implement these later.

Reply:
yes, as of now, only version command that currently using it. Just for the example. Will put TODO comment here to avoid confusion